File: LeetCode/150. Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        int n = tokens.size();
        for (string c : tokens) {
            if (c == "+") {
                int a = st.top(); st.pop();
                int b = st.top(); st.pop();
                st.push(a + b);
            } else if (c == "-") {
                int a = st.top(); st.pop();
                int b = st.top(); st.pop();
                st.push(b - a);
            } else if (c == "*") {
                int a = st.top(); st.pop();
                int b = st.top(); st.pop();
                st.push(a * b);
            } else if (c == "/") {
                int a = st.top(); st.pop();
                int b = st.top(); st.pop();
                st.push(b / a);
            } else {
                st.push(stoi(c));
            }
        }
        return st.top();
    }
};

File: neetcode-150/stack/150. Evaluate Reverse Polish Notation.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        Stack<int> st = new Stack<int>();
        int n = tokens.Length;
        foreach (string c in tokens) {
            if (c == "+") {
                int a = st.Peek(); st.Pop();
                int b = st.Peek(); st.Pop();
                st.Push(a + b);
            } else if (c == "-") {
                int a = st.Peek(); st.Pop();
                int b = st.Peek(); st.Pop();
                st.Push(b - a);
            } else if (c == "*") {
                int a = st.Peek(); st.Pop();
                int b = st.Peek(); st.Pop();
                st.Push(a * b);
            } else if (c == "/") {
                int a = st.Peek(); st.Pop();
                int b = st.Peek(); st.Pop();
                st.Push(b / a);
            } else {
                st.Push(int.Parse(c));
            }
        }
        return st.Peek();
    }
}